strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/metadata/MavenSnapshotManager.java

```diff
@@ -192,41 +183,42 @@ private boolean removeTimestampedSnapshot(RepositoryPath basePath,
                 .getSnapshotVersions()
                 .forEach(e ->
                          {
-
                              if ("jar".equals(e.getExtension()))
                              {
-                                 String version = e.getVersion();
-                                 snapshots.put(Integer.parseInt(ArtifactUtils.getSnapshotBuildNumber(version)),
-                                               version);
+                                 SnapshotVersionDecomposition snapshotVersion = SnapshotVersionDecomposition.of(e.getVersion());
+                                 if (!SnapshotVersionDecomposition.INVALID.equals(snapshotVersion))
```

Comment:
@fuss86 maybe we should have some **WARNING** logs in case of `else`?

Reply:
ok, will do